src/three-components/SmoothControls.ts

```diff
@@ -338,10 +364,31 @@ export class SmoothControls extends EventDispatcher {
       boolean {
     const {theta, phi, radius} = this[$targetSpherical];
 
-    const targetTheta = theta - deltaTheta;
-    const targetPhi = phi - deltaPhi;
+    let targetPhi = phi - deltaPhi;
     const targetRadius = radius + deltaRadius;
 
+    const {
+      minimumPolarAngle: minPhi,
+      maximumPolarAngle: maxPhi,
+    } = this[$options];
+
+    let phiScale = 1;
+
+    if (targetPhi < minPhi!) {
+      phiScale = 1 - Math.min((targetPhi - minPhi!) / (this[$minOverSpherical].phi - minPhi!), 1);
+    }
+
+    if (targetPhi > maxPhi!) {
+      phiScale = 1 - Math.min((targetPhi - maxPhi!) / (this[$maxOverSpherical].phi - maxPhi!), 1);
```

Comment:
The math here doesn't seem quite right. if phi = maxPhi, and I command a deltaPhi that would take me past maxOverSpherical, this would change that delta to zero, when I would instead expect it to be clamped. I think you want a mapping that takes the input delta: [0, inf] and maps it asymptotically to an output delta of [0, overRotateAmount] (see my comment below).